src/Riok.Mapperly.Abstractions/MapperAttribute.cs

```diff
@@ -28,6 +28,23 @@ public sealed class MapperAttribute : Attribute
     /// </summary>
     public bool EnumMappingIgnoreCase { get; set; }
 
+    /// <summary>
+    /// Specifies the behaviour in the case when the mapper tries to return <c>null</c> in a mapping method with a non-nullable return type.
+    /// If set to <c>true</c> an <see cref="ArgumentNullException"/> is thrown.
+    /// If set to <c>false</c> the mapper tries to return a default value.
+    /// For a <see cref="string"/> this is <see cref="string.Empty"/>,
+    /// for value types <c>default</c>
+    /// and for reference types <c>new()</c> if a parameterless constructor exists or else an <see cref="ArgumentNullException"/> is thrown.
```

Comment:
Shouldn't this last case also emit a diagnostic?

Reply:
Added a diagnostic.